File: Src/KartuParkir.php
<?php
class KartuParkir {
    private string $nomor;
    private string $jenis;
    private string $status;

    public function __construct($nomor, $jenis) {
        if (empty($nomor))
            throw new InvalidArgumentException("Nomor kartu kosong!");
        $this->nomor = $nomor;
        $this->jenis = $jenis;
        $this->status = "Belum Masuk";
    }

    public function masuk() {
        if ($this->status != "Belum Masuk")
            throw new InvalidArgumentException("Kendaraan sudah masuk!");
        $this->status = "Aktif";
        echo "Kendaraan berhasil masuk.\n";
    }

    public function keluar() {
        if ($this->status != "Aktif")
            throw new InvalidArgumentException("Kendaraan belum masuk!");
        $this->status = "Selesai";
        echo "Kendaraan berhasil keluar.\n";
    }

    public function cekStatus() {
        echo "Kartu: $this->nomor\nKendaraan: $this->jenis\nStatus: $this->status\n";
    }
}

File: Src/KartuParkir.java
public class KartuParkir {
    String nomorKartu;
    String jenisKendaraan;
    String status;

    public KartuParkir(String nomor, String jenis) {
        if (nomor.isEmpty())
            throw new IllegalArgumentException("Nomor kartu kosong!");

        nomorKartu = nomor;
        jenisKendaraan = jenis;
        status = "Belum Masuk";
    }

    public void masuk() {
        if (status.equals("Aktif"))
            throw new IllegalArgumentException("Kartu sudah digunakan!");

        if (status.equals("Selesai"))
            throw new IllegalArgumentException("Kartu sudah selesai digunakan!");

        status = "Aktif";
        System.out.println("Kendaraan berhasil masuk.");
    }

    public void keluar() {
        if (!status.equals("Aktif"))
            throw new IllegalArgumentException("Kendaraan belum berada di area parkir!");

        status = "Selesai";
        System.out.println("Kendaraan berhasil keluar.");
    }

    public void cekStatus() {
        System.out.println("Nomor Kartu : " + nomorKartu);
        System.out.println("Kendaraan   : " + jenisKendaraan);
        System.out.println("Status      : " + status);
    }
}